variables.tf

```
@@ -15,36 +15,91 @@ variable "hosted_zone" {
   description = "Amazon Route 53 hosted zone name."
   type        = string
   validation {
-    condition     = trim(var.hosted_zone, " ") != ""
+    condition     = length(trimspace(var.hosted_zone)) > 0
     error_message = "Host name must not be an empty string."
   }
+  validation {
+    condition     = can(regex("^([a-zA-Z0-9-]+\\.)+[a-zA-Z]+$", var.hosted_zone))
+    error_message = "Host name must be a valid domain name."
+  }
 }
 
 variable "cert_arn" {
   description = "AWS Certificate Manager (ACM) certificate for Amazon Resource Names (ARN)."
   type        = string
 
   validation {
-    condition     = can(regex("^arn", var.cert_arn))
-    error_message = "The cert_arn should start with ARN."
+    condition     = can(regex("^arn:aws:acm:", var.cert_arn))
+    error_message = "The cert_arn should be a valid ACM certificate ARN."
+  }
+  validation {
+    condition     = length(var.cert_arn) > 0
+    error_message = "The cert_arn must not be an empty string."
   }
 }
 
 variable "trial_license" {
   description = "CloudBees CI trial license details for evaluation."
   type        = map(string)
+  validation {
+    condition     = contains(keys(var.trial_license), "first_name") && contains(keys(var.trial_license), "last_name") && contains(keys(var.trial_license), "email") && contains(keys(var.trial_license), "company")
+    error_message = "The trial_license must contain the following keys: first_name, last_name, email, company."
+  }
+  validation {
+    condition     = length(var.trial_license) == 4
+    error_message = "The map must contain 4 keys."
+  }
 }
 
-variable "create_k8s_secrets" {
-  description = "Create the Kubernetes cbci-secrets. It can be consumed by CloudBees CasC for the operations center."
+variable "create_casc_secrets" {
+  description = "Create a Kubernetes basic secret for CloudBees Configuration as Code (cbci-sec-casc) and mount it into the Operation Center /var/run/secrets/cbci."
   default     = false
   type        = bool
 }
 
-variable "k8s_secrets" {
-  description = "Secrets .yml file as a string containing the names:values secrets. It is required when create_k8s_secrets is enabled."
+variable "casc_secrets_file" {
+  description = "Secrets .yml file path containing the names:values secrets. It is required when create_casc_secrets is enabled."
   default     = "secrets-values.yml"
   type        = string
+  validation {
+    condition     = length(trimspace(var.casc_secrets_file)) > 0
+    error_message = "Casc secret file  must not be an empty string."
+  }
+}
+
+variable "create_reg_secret" {
+  description = "Create a Kubernetes dockerconfigjson secret for Container Registry authentication (cbci-sec-reg) for CI builds agents."
```

Is dockerconfigjson correct?
```suggestion
  description = "Create a Kubernetes dockerconfigjson secret for container registry authentication (cbci-sec-reg) for CI builds agents."
```